portal-impl/src/com/liferay/portal/systemevent/SystemEventAdvice.java

```diff
@@ -182,7 +187,10 @@ protected void duringFinally(
 	protected String getClassName(ClassedModel classedModel) {
 		String className = classedModel.getModelClassName();
 
-		if (classedModel instanceof StagedModel) {
+		if ((classedModel instanceof StagedModel) &&
+			!StringUtil.startsWith(
```

Comment:
I would like to know if it is possible to override the `getStagedModelType` method of the `ObjectEntry` implementation. If so, also would like to know if there is something risky on doing that